lab_2_tokenize_by_bpe/main.py

```diff
@@ -26,6 +36,20 @@ def collect_frequencies(
     :param end_of_word: a token that signifies the end of word
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if (not isinstance(text, str)
+            or not isinstance(start_of_word, str | None)
+            or not isinstance(end_of_word, str)):
+        return None
+    frequencies = {}
+    for word in text.split(' '):
+        word_key = prepare_word(word, start_of_word, end_of_word)
+        if word_key is None:
+            return None
+        if word_key in frequencies:
```

Comment:
simplify - remove else (set 0 if the key is not in dict)

Reply:
fixed